config/cmake/HDFMacros.cmake

```diff
@@ -16,8 +16,13 @@ macro (SET_HDF_BUILD_TYPE)
   if (_isMultiConfig)
     set_property (CACHE CMAKE_CONFIGURATION_TYPES PROPERTY STRINGS "Debug" "Release"
       "MinSizeRel" "RelWithDebInfo" "Developer" FORCE)
-    set (HDF_CFG_NAME "RelWithDebInfo")
-    set (HDF_BUILD_TYPE ${CMAKE_CFG_INTDIR})
+    if (CMAKE_BUILD_TYPE)
+      set (HDF_CFG_NAME ${CMAKE_BUILD_TYPE})
+      set (HDF_BUILD_TYPE ${CMAKE_BUILD_TYPE})
```

Comment:
Is the difference between `HDF_CFG_NAME` and `HDF_BUILD_TYPE` documented somewhere?

Reply:
One of the things I was doing in the patch I was putting together included a comment block where I documented things like this. I think it's something that needs to be done, but I don't think that should hold this PR.